File: src/main/kotlin/aurocosh/divinefavor/common/network/message/client/syncing/MessageSyncFlyingCapability.java
package aurocosh.divinefavor.common.network.message.client.syncing;

import aurocosh.divinefavor.DivineFavor;
import aurocosh.divinefavor.common.network.base.WrappedClientMessage;
import net.minecraft.entity.player.EntityPlayer;
import net.minecraftforge.fml.relauncher.Side;
import net.minecraftforge.fml.relauncher.SideOnly;

public class MessageSyncFlyingCapability extends WrappedClientMessage {
    public boolean canFly;

    public MessageSyncFlyingCapability() {
    }

    public MessageSyncFlyingCapability(boolean canFly) {
        this.canFly = canFly;
    }

    @Override
    @SideOnly(Side.CLIENT)
    protected void handleSafe() {
        EntityPlayer player = DivineFavor.proxy.getClientPlayer();
        player.capabilities.allowFlying = canFly;
        player.capabilities.isFlying = canFly;
    }
}

File: src/main/kotlin/aurocosh/divinefavor/common/network/message/client/syncing/MessageSyncFlyingCapability.kt
package aurocosh.divinefavor.common.network.message.client.syncing

import aurocosh.divinefavor.DivineFavor
import aurocosh.divinefavor.common.network.message.base.DivineClientMessage
import net.minecraftforge.fml.relauncher.Side
import net.minecraftforge.fml.relauncher.SideOnly

class MessageSyncFlyingCapability : DivineClientMessage {
    var canFly: Boolean = false

    constructor() {}

    constructor(canFly: Boolean) {
        this.canFly = canFly
    }

    @SideOnly(Side.CLIENT)
    override fun handleSafe() {
        val player = DivineFavor.proxy.clientPlayer
        player.capabilities.allowFlying = canFly
        player.capabilities.isFlying = canFly
    }
}
